### ISSUE TITLE
Support for using existing distribution/development certificates with fast lane setup

### ISSUE THREAD

**Issue by nareshsvs** [fastlane/fastlane#5737]
Currently fast lane generates new certificates and profiles and cannot work with existing distribution certs & profiles. 

I am able to come close to using this by generating new provisioning profiles and use existing certificates with few changes to existing api.

- API to encrypt and decrypt repo outside of fast lane actions by using method in encrypt.rb of match library
- Few custom actions to add files and remove files to repo , so that we can iteratively add files to the repository.
- made changes to match params to take p12 password as a parameter so that certificates can be imported silently.

Is this a relevant feature for everyone or do you suggest keeping this version custom to our deployment. One of my concerns of having this local is, we will not be able to take new changes easily.

Could you please suggest a path forward, if this can be added as an enhancement request to the tools.

**Comment by nareshsvs** [comment]
No it is not code signing problem, current tool doesn't allow users with existing certificates and profiles to adopt fastlane easily unless new certificates and profiles are generated.

**Comment by TKBurner** [comment]
@nareshsvs Thanks for following up. What specific tools are you trying to use in Fastlane with the pre-existing provisioning profiles? Thanks!

**Comment by nareshsvs** [comment]
I am primarily using match functionality for renewing profiles

**Comment by KrauseFx** [comment]
Yah, we've had the discussion a few times already. We decided not to offer an import feature for existing profiles and certificates. The reason for that is that users might import broken certificates, or profiles that don't match the certificate. The beauty of match is that it takes care of all those things for you and makes sure everything works perfectly. match also makes some assumptions around how profiles are named, so it might not even work flawlessly. 
Thanks for the suggestion though, but we decided not to offer this feature for now 👍
